### PR TITLE
HDDS-5511. ContainerStateMap should handle ecReplication config map

### PR DESCRIPTION
**Pull request by sodonnel**
## What changes were proposed in this pull request?

Continue replacing instances of ReplicationType and Factor with Replication Config in the ContainerStateMap and other impacted classes.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5511

## How was this patch tested?

Existing tests

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/states/ContainerStateMap.java [hunk 1]
```diff
@@ -85,7 +84,7 @@
 
   private final ContainerAttribute<LifeCycleState> lifeCycleStateMap;
   private final ContainerAttribute<String> ownerMap;
-  private final ContainerAttribute<ReplicationFactor> factorMap;
+  private final ContainerAttribute<ReplicationConfig> factorMap;
```
Comment (umamaheswararao) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/states/ContainerStateMap.java:87]: you may want to rename factorMap to replicationConfigMap ?

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/states/ContainerStateMap.java [hunk 2]
```diff
@@ -85,7 +84,7 @@
```
Comment (umamaheswararao) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/states/ContainerStateMap.java:84]: You may want to update the [documentation](https://github.com/apache/ozone/blob/e7e984c559dabe648f4ab81d0f7d901ffcc50fe6/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/states/ContainerStateMap.java#L65) of this class. 

Currently it's talking about factor.

**Review by umamaheswararao — COMMENTED** [review]
Thanks @sodonnel for the patch. I have added few minors. Please take a look at it.

**Comment by sodonnel** [comment]
@umamaheswararao Thanks for the review comments. I have pushed another commit to address your suggestions. Please have a look once the Ci comes back green.

**Review by umamaheswararao — APPROVED** [review]
Latest changes looks good to me. Thanks

**Pull request merged**